src/git.rs

```diff
@@ -638,6 +639,10 @@ async fn fetch_remote(args: &[&str]) -> Result<Vec<RemoteInfo>> {
                 .unwrap_or_else(|| "None".into())
         );
     }
+    log::debug!("git ls-remote stdout:");
+    String::from_utf8_lossy(&process.stdout)
```

Comment:
While this probably works it wouldn't help a lot of `git` is stuck at some point during the execution. Since this is already `async` we could log lines as they appear? Probably something for the future but I'm usually deeply frustrated by applications that eat the error output (we only provide stdout) and that block until the subprocess is finished. Usually I just attach `strace` and check again, but nobody should be required to resort to that.

Reply:
I can look into it in a follow up. Ideally we'd factor out the process of shelling out into a helper function which does the appropriate wrapping, so that id doesn't have to be repeated at every call site (even though currently there are only two). In the case of git specifically, we could maybe also look into using libgit or any other Rust git bindings to call the code directly